pkg/actuators/machine/reconciler.go

```diff
@@ -211,6 +211,13 @@ func (r *Reconciler) update() error {
 	runningLen := len(runningInstances)
 	var newestInstance *ec2.Instance
 
+	//Prepare the tag list with infrastructure tags.
+	//these tags will update to the ec2 instance.
+	tagList, err := r.getTagsFromInfrastructure()
+	if err != nil {
+		return err
```

Comment:
We tend to add extra context to errors to help debugging, adding a unique string helps identify which function returned the error
```suggestion
		return fmt.Errof("error fetching Infrastructure tags: %v", err)
```

Reply:
I feel, error wrapping at line 265 is good , else it will wrap multiple error stack.